### ISSUE TITLE
TornLifestyle warning caused by decorator registration

### ISSUE THREAD

**Issue by kalderman** [simpleinjector/SimpleInjector#24]
I have a class implementing multiple role-interfaces and would like to decorate one of those role interfaces.

```C#
public interface IA { }
public interface IB { }

public class ABComposite : IA, IB { }
public class BDecorator : IB 
{
    public BDecorator(IB b) { }
}
```

I register ABComposite like this:
```C#
 var container = new Container();

var registration = Lifestyle.Singleton.CreateRegistration<ABComposite>(container);
container.AddRegistration(typeof(IA), registration);
container.AddRegistration(typeof(IB), registration);
```
This registration passes the following unit test:
```C#
container.Verify();

var results = Analyzer.Analyze(container);
Assert.Empty(results);
```
Then I register the decorator like this:
```C#
container.RegisterSingleDecorator(typeof(IB), typeof(BDecorator));
```
Now the unit test fails because results have the following TornLifeStyle warnings:

>IA: The registration for IA maps to the same implementation and lifestyle as the registration for IB does. They both map to ABComposite (Singleton). This will cause each registration to resolve to a different instance: each registration will have its own instance.
>IB: The registration for IB maps to the same implementation and lifestyle as the registration for IA does. They both map to ABComposite (Singleton). This will cause each registration to resolve to a different instance: each registration will have its own instance.

It turns out that only one instance of ABComposite is created.  This is the desired behavior, so I believe the container is working correctly, however the Analyzer is not.

**Comment by dotnetjunkie** [comment]
I can confirm this. This is a bug in the torn lifestyle warning. The bug exists both in v2.7.3 and v2.8.0-beta1. I'll take a look if we can fix this in v2.8.

**Comment by dotnetjunkie** [comment]
We rolled out v2.8.0-beta2 to NuGet that fixes this issue.

**Comment by kalderman** [comment]
Confirming this is fixed in v2.8.0-beta2.
Thanks!

**Comment by dotnetjunkie** [comment]
Based on your response I conclude that our new [Ambiguous Lifestyles warning](https://github.com/simpleinjector/Documentation/blob/2.8/source/ambiguouslifestyles.rst) isn't causing you any trouble :-)

**Comment by kalderman** [comment]
Nope. :)

**Comment by dotnetjunkie** [comment]
Fixed in v2.8.
